spec/integration_spec.rb

```diff
@@ -299,6 +299,76 @@ def parse_html(html)
       expect(result).to include("<td>One</td><td>1</td>")
       expect(result).to include("<td>Two</td><td>2</td>")
     end
+
+    it "doesn't inline styles in media queries with features" do
+      document = Roadie::Document.new <<-HTML
+        <html>
+          <head>
+            <link rel="stylesheet" href="/style.css">
+          </head>
+          <body>
+            <div class="colorful"></div>
+          </body>
+        </html>
+      HTML
+
+      document.asset_providers = TestProvider.new(
+        "/style.css" => <<-CSS
+          .colorful { color: green; }
+          @media screen and (max-width 600px) {
+            .colorful { color: red; }
+          }
+        CSS
+      )
+
+      result = parse_html document.transform
+      expect(result).to have_styling('color' => 'green').at_selector('.colorful')
+    end
+
+    it "should put non-inlineable media queries in the head" do
+      document = Roadie::Document.new <<-HTML
+        <html>
+          <head>
+            <link rel="stylesheet" href="/style.css">
+          </head>
+          <body>
+            <div class="colorful"></div>
+          </body>
+        </html>
+      HTML
+
+      document.asset_providers = TestProvider.new(
+        "/style.css" => <<-CSS
+          .colorful { color: green; }
+          @media screen and (max-width 600px) {
+            .colorful { color: red; width: 600px; }
+            .colorful-2 { color: red; width: 600px; }
+          }
+          @media screen and (max-width 800px) {
+            .colorful { color: blue; }
+          }
+          @media screen, print and (max-width 800px) {
+            .colorful { color: blue; }
+          }
```

Comment:
I want a test for the optimization that merges media queries, as well as a test that looks at a very simple case. I suggest changing this into the optimization test (split the top media query into two and place one half at the end).

Then add another one with a single media query before this test that can show the feature off.

Thank you.